Rules/CommonScripts/GameStateBanners.as

```diff
@@ -0,0 +1,189 @@
+#define CLIENT_ONLY
+
+Vec2f bannerStart = Vec2f_zero;
+Vec2f bannerPos = Vec2f_zero;
+Vec2f bannerDest = Vec2f_zero;
+f32 frameTime = 0;
+const f32 maxTime = 0.6f;
+
+bool minimap = true;
+
+const u32 winBannerDuration = 8 * getTicksASecond();
+const u32 buildBannerDuration = 5 * getTicksASecond();
+const u32 gameBannerDuration = 5 * getTicksASecond();
+
+void onInit(CRules@ this)
+{
+	if (!GUI::isFontLoaded("AveriaSerif-Bold_32"))
+	{
+		string AveriaSerif = CFileMatcher("AveriaSerif-Bold.ttf").getFirst();
+		GUI::LoadFont("AveriaSerif-Bold_32", AveriaSerif, 32, true);
+	}
+
+	minimap = this.minimap;
+	onRestart(this);
+}
+
+void onRestart(CRules@ this)
+{
+	ResetBannerInfo(this);
+	this.minimap = minimap;
+
+	SetBanner(this);
+}
+
+void onNewPlayerJoin(CRules@ this, CPlayer@ player)
+{
+	if (getLocalPlayer() !is null && getLocalPlayer().getUsername() == player.getUsername())
+	{
+		SetBanner(this);
+	}
+}
+
+void onStateChange(CRules@ this, const u8 oldState)
+{
+	ResetBannerInfo(this);
+	SetBanner(this);
+}
+
+void onTick(CRules@ this)
+{
+	if (this.get_string("Animate Banner") != "none" && this.get_u32("Banner End") < getGameTime())
+	{
+		ResetBannerInfo(this);
+	}
+}
+
+void onRender(CRules@ this)
+{
+	if (this.get_string("Animate Banner") != "none")
+	{
+		Driver@ driver = getDriver();
+		if (driver !is null)
+		{
+			if (bannerPos != bannerDest)
+			{
+				frameTime = Maths::Min(frameTime + (getRenderDeltaTime() / maxTime), 1);
+
+				bannerPos = Vec2f_lerp(bannerStart, bannerDest, frameTime);
+			}
+
+			if (this.get_string("Animate Banner") == "win") 
+			{
+				DrawWinBanner(bannerPos, this.getTeamWon());
+				this.SetGlobalMessage("");
+			}
+			// todo: implement tth versions (alternatively remove TTH)
+			else if (this.get_string("Animate Banner") == "build" && (this.gamemode_name == "CTF" || this.gamemode_name == "SmallCTF"))
+			{
+				DrawBuildBanner(bannerPos);
+			} 
+			else if (this.get_string("Animate Banner") == "game" && (this.gamemode_name == "CTF" || this.gamemode_name == "SmallCTF"))
+			{
+				CPlayer@ p = getLocalPlayer();
+				int team = p is null ? 0 : p.getTeamNum();
+				// show flags of enemy team colour
+				if (team == 0) team = 1;
+				else team = 0;
+
+				DrawGameBanner(bannerPos, team);
+			} 
+		}
+	}
+}
+
+void ResetBannerInfo(CRules@ this)
+{
+	this.set_string("Animate Banner", "none");
+	bannerPos = Vec2f_zero;
+	bannerDest = Vec2f_zero;
+	frameTime = 0;
```

Comment:
`0.0f` :kag_sweat:

And there is no point in resetting bannerPos and bannerDest (see start of file review comment)

Reply:
fixed